stdlib/LinearAlgebra/src/lq.jl

```diff
@@ -70,7 +70,7 @@ julia> l == S.L &&  q == S.Q
 true
 ```
 """
-lq(A::StridedMatrix{<:BlasFloat})  = lq!(copy(A))
+lq(A::AbstractMatrix)  = lq!(copy(A))
 lq(x::Number) = lq(fill(x,1,1))
```

Comment:
```suggestion
lq(A::AbstractMatrix{T}) where {T}  = lq!(copy_oftype(A, lq_eltype(T)))
lq(x::Number) = lq!(fill(x, 1, 1))

lq_eltype(::Type{T}) where {T} = typeof(zero(T) / sqrt(abs2(one(T))))
```

Reply:
I'm assuming merging meant you approved, so I marked the PR mergeme, for when CI finishes, but feel free to clear that if you have concerns.

Reply:
ah, needs a rebase too, to pick up CI updates (@oxinabox)